### PR TITLE
U/dc/context and particles

### PR DESCRIPTION
**Pull request by mtrent6**
* Adds Song List next to Particles
* Adds raycasting to particles
* still need to sync list and particles together

### CODE REVIEW

**Review by afitz4 — COMMENTED** [review]
this is a unit of a pr Ω

File: frontend/src/components/SongList/SongList.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+interface ISongListState {
+    songs: Song[];
+    showList: boolean;
+}
+
+export const SongList: React.FC<ISongList & ISongListProps> = (props: ISongListProps) => {
+    const [songs, setSongs] = useState(props.songs);
+    // const [song, setSong] = useState(props.song)
+    const [showList, setShowList] = useState(false);//???????
```
Comment (Cutaiar) [inline frontend/src/components/SongList/SongList.tsx:26]: ?????

File: frontend/src/components/Spotiverse/Spotiverse.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const setSong = (song: Song) => {
+    setSelectedSong(song)
+  }
+
+  return (
+    <>
+      <div style={{ display: "flex", flexDirection: "row", height: "87vh" }}>
+        <ThreeEngine songs={TestData} setSong={setSong} song={selectedSong} />
+        <div style={{ color: "white" }} className="songListWrapper">
+          <SongList song={selectedSong} setSong={setSong} songs={TestData} />
```
Comment (Cutaiar) [inline frontend/src/components/Spotiverse/Spotiverse.tsx:44]: nit: `selectedSong`, `setSelectedSong`

**Review by Cutaiar — APPROVED** [review]
awesome change man

File: frontend/src/components/Accordion/Accordion.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    return (
+        <div id={props.id} className="accordion__section" >
+            <button className={`accordion ${active}`} onClick={toggleAccordion} >
+                {props.mainItem}
+                < Chevron className={`${setRotate}`} width={10} fill={"#777"} />
+            </button>
+            < div
+                ref={content}
+                style={{ maxHeight: `${setHeight}` }}
+                className="accordion__content"
```
[the remark this answers is not in the document]
Comment (mtrent6) [reply frontend/src/components/Accordion/Accordion.tsx:50]: tbh i found this code online

File: frontend/src/components/SongList/SongList.tsx (hunk 1 above)
Comment (mtrent6) [reply frontend/src/components/SongList/SongList.tsx:26]: i completely forget why I added the question marks

**Pull request merged**